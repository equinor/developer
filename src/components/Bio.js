import React, { Component } from "react";
import kebabCase from "lodash/kebabCase";
import styled from "styled-components";
import Image from "gatsby-image";

const Wrapper = styled.div`
  display: flex;
  margin-bottom: 0;
  flex-direction: row;
  @media (max-width: 576px) {
    flex-direction: column;
    margin-bottom: 20px;
  }
`;

const AuthorContent = styled.div`
  display: flex;
  margin-right: 10px;
  flex-direction: column;
`;

const Avatar = styled.img`
  margin-bottom: 0;
  margin-right: 10px;
  width: 40px;
  height: 40px;
  border-radius: 100%;
`;

class Authors extends Component {
  render() {
    const { authors } = this.props;

    const list = authors && authors.map((author, index) => {
      return <Bio key={`${author}-${index}`} author={author} />;
    });

    return <Wrapper>{list}</Wrapper>;
  }
}

export { Authors };

class Bio extends Component {
  render() {
    const { author } = this.props;

    const { name, image } = author;

    return (
      <Wrapper>
        <Avatar
          src={image}
          alt={name}
          imgStyle={{
            borderRadius: `50%`,
          }}
        />
        <AuthorContent>
          <div className="bio-name">{`${name}`}</div>
          {false && (
            <div className="bio-email">
              <a href={`mailto:${email}`}>{email}</a>
            </div>
          )}
          {false && <div className="bio-bio">{bio}</div>}
        </AuthorContent>
      </Wrapper>
    );
  }
}

export default Bio;
