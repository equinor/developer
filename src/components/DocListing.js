import React from "react";
import { Link } from "gatsby";
import { Tag } from "./Tags";
import { Divider } from "./BlogListing";
import { getIcon } from "../util/iconUtil";
import { Col } from "react-styled-flexboxgrid";

export default ({ node }) => {
  const {
    frontmatter: { title, tags },
    fields: { slug, collection },
  } = node;
  //find base slug.
  const Icon = getIcon(tags[0]);
  return (
    <Col xs={12} md={10} mdOffset={1}>
      <div style={{}}>
        <Icon style={{ height: 30, width: 30 }} />
        <Tag tag={tags[0]} />
      </div>
      <Link to={`/${collection}${slug}`} style={{ padding: "10px 0 20px" }}>
        {title}
      </Link>
      <Divider />
    </Col>
  );
};
