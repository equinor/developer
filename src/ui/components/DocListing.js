import React from "react";
import { Link } from "gatsby";
import { getIcon } from "./Categories";
import { Tag } from "./Tags";
import { Divider } from "../../components/BlogListing";

export default ({ node }) => {
  const {
    frontmatter: { title },
    fields: { slug, collection },
  } = node;
  //find base slug.
  const category = slug.substr(1, slug.substr(1).indexOf("/"));
  const Icon = getIcon(node.fields);
  return (
    <React.Fragment>
      <div style={{}}>
        <Icon style={{ height: 30, width: 30 }} />
        <Tag tag={category} />
      </div>
      <Link to={`/${collection}${slug}`} style={{ padding: "10px 0 20px" }}>
        {title}
      </Link>
      <Divider />
    </React.Fragment>
  );
};
