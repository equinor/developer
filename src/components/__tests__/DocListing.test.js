import React from "react";
import { mockAuthors } from "./Bio.test";
import renderer from "react-test-renderer";
import DocListing from "../DocListing";

describe("BlogListing", () => {
  const node = {
    frontmatter: { title: "Api", tags: ["Api"] },
    fields: { authors: mockAuthors },
  };

  it("renders correctly", () => {
    const tree = renderer.create(<DocListing node={node} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
