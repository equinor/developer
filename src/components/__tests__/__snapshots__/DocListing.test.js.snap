// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bio renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: 10px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  margin-bottom: 0;
  margin-right: 10px;
  width: 40px;
  height: 40px;
  border-radius: 100%;
}

@media (max-width:576px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    margin-bottom: 20px;
  }
}

<div
  className="c0"
>
  <div
    className="c0"
  >
    <img
      alt="Ola Nordmann"
      className="c1"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAaQAAAGkCAIAAADxLsZiAAAF4klEQVR4nOzXAY3jMBBA0dtTiJRWyEQhE1qhYAjLoeraav57BGakxl/TbYzxD+Dp/q9eAGAGsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxK2aZOu8542i7ftx2v1Cp/kq/sKc746lx2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkDCNm3SfrymzZrjOu/VK1D0vKc0h8sOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5I+BljrN4B4M+57IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IGGbNuk672mzptmP1+oVaPGO3uayAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDErbVC3y367xXr/Bh+/FavcInPe8H4m3zYvewV+QhscrzntIc/sYCCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJP2OM1TsA/DmXHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkPAbAAD//9JGJL1D6GKKAAAAAElFTkSuQmCC"
    />
    <div
      className="c2"
    >
      <div
        className="bio-name"
      >
        Ola Nordmann
      </div>
    </div>
  </div>
  <div
    className="c0"
  >
    <img
      alt="Kari Nordmann"
      className="c1"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAaQAAAGkCAIAAADxLsZiAAAF4klEQVR4nOzXAY3jMBBA0dtTiJRWyEQhE1qhYAjLoeraav57BGakxl/TbYzxD+Dp/q9eAGAGsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxK2aZOu8542i7ftx2v1Cp/kq/sKc746lx2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkDCNm3SfrymzZrjOu/VK1D0vKc0h8sOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5I+BljrN4B4M+57IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IGGbNuk672mzptmP1+oVaPGO3uayAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDErbVC3y367xXr/Bh+/FavcInPe8H4m3zYvewV+QhscrzntIc/sYCCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJP2OM1TsA/DmXHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkPAbAAD//9JGJL1D6GKKAAAAAElFTkSuQmCC"
    />
    <div
      className="c2"
    >
      <div
        className="bio-name"
      >
        Kari Nordmann
      </div>
    </div>
  </div>
  <div
    className="c0"
  >
    <img
      alt="Erik Eriksen"
      className="c1"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAaQAAAGkCAIAAADxLsZiAAAF4klEQVR4nOzXAY3jMBBA0dtTiJRWyEQhE1qhYAjLoeraav57BGakxl/TbYzxD+Dp/q9eAGAGsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxK2aZOu8542i7ftx2v1Cp/kq/sKc746lx2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkDCNm3SfrymzZrjOu/VK1D0vKc0h8sOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5I+BljrN4B4M+57IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IGGbNuk672mzptmP1+oVaPGO3uayAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDErbVC3y367xXr/Bh+/FavcInPe8H4m3zYvewV+QhscrzntIc/sYCCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJP2OM1TsA/DmXHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkPAbAAD//9JGJL1D6GKKAAAAAElFTkSuQmCC"
    />
    <div
      className="c2"
    >
      <div
        className="bio-name"
      >
        Erik Eriksen
      </div>
    </div>
  </div>
</div>
`;

exports[`BlogListing renders correctly 1`] = `
.c2 {
  font-size: 12px;
  font-weight: 500;
  line-height: 14px;
  cursor: inherit;
  color: #007079;
  text-transform: uppercase;
}

.c0 {
  box-sizing: border-box;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding-right: 0.5rem;
  padding-left: 0.5rem;
}

.c3 {
  border-bottom: 1px solid #f2f2f2;
  margin: 20px 0;
}

.c1 {
  font-size: 25px;
  color: black;
  font-weight: 200;
  padding: 0 2px 0 10px;
}

@media only screen and (min-width:0em) {
  .c0 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    max-width: 100%;
    display: block;
  }
}

@media only screen and (min-width:64em) {
  .c0 {
    -webkit-flex-basis: 83.33333333333334%;
    -ms-flex-preferred-size: 83.33333333333334%;
    flex-basis: 83.33333333333334%;
    max-width: 83.33333333333334%;
    display: block;
  }
}

@media only screen and (min-width:64em) {
  .c0 {
    margin-left: 8.333333333333334%;
  }
}

<div
  className="c0"
>
  <div
    style={Object {}}
  >
    <test-file-stub
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
    <span
      className="c1"
    >
      /
    </span>
    <span
      className="c2"
    >
      Api
    </span>
  </div>
  <a
    href="/undefinedundefined"
    style={
      Object {
        "padding": "10px 0 20px",
      }
    }
  >
    Api
  </a>
  <div
    className="c3"
  />
</div>
`;
